visidata/loaders/xlsx.py

```diff
@@ -28,33 +31,52 @@ def iterload(self):
             src = self.workbook[sheetname]
             yield XlsxSheet(self.name, sheetname, source=src)
 
-
 class XlsxSheet(SequenceSheet):
     # rowdef: AttrDict of column_letter to cell
     def setCols(self, headerrows):
+        from openpyxl.utils.cell import get_column_letter
         self.columns = []
         self._rowtype = AttrDict
 
         if not headerrows:
             return
 
-        headers = [[x.value for x in row.values()] for row in headerrows]
-        column_letters = [x.column_letter for x in headerrows[0].values()]
+        headers = [[cell.value for cell in row.values()] for row in headerrows]
+        column_letters = [
+                x.column_letter if 'column_letter' in dir(x)
+                else get_column_letter(i+1)
+                for i, x in enumerate(headerrows[0].values())]
 
         for i, colnamelines in enumerate(itertools.zip_longest(*headers, fillvalue='')):
             colnamelines = ['' if c is None else c for c in colnamelines]
-            self.addColumn(AttrColumn(''.join(map(str, colnamelines)), column_letters[i] +'.value'))
+            column_name = ''.join(map(str, colnamelines))
+            self.addColumn(AttrColumn(column_name, column_letters[i] + '.value'))
+            self.addXlsxColumns(column_letters[i], column_name)
 
     def addRow(self, row, index=None):
         Sheet.addRow(self, row, index=index)  # skip SequenceSheet
         for column_letter, v in list(row.items())[len(self.columns):len(row)]:  # no-op if already done
-            self.addColumn(AttrColumn('', column_letter+'.value'))
+            self.addColumn(AttrColumn(column_letter, column_letter + '.value'))
```

Comment:
The reason we used `''` and let it auto-assign the column letter, is that the column name is officially "blank" for purposes of `g^` (which sets blank column names from rows). This might set the column letter different from the actual spreadsheet, but `g^` is important for any table where the headers may be part of the data.

Reply:
Got it. What about the following line? Should I be leaving those column names blank too?

Reply:
Yes, I'd leave all the names for the basic value columns blank.  The color/etc additional column should keep their names.